src/utils.jl

```diff
@@ -1,10 +1,48 @@
+using ExactOptimalTransport, HiGHS
 
 function normalized_exp!(weight::AbstractVector)
     weight .-= maximum(weight)
     @. weight = exp(weight)
     weight ./= sum(weight)
 end
 
+# Solve an optimal transport problem to minimize the number of particles
+# that need to be communicated between ranks during resampling.
+function optimized_resample!(resampled_indices::AbstractVector{Int}, nrank::Int)
+    nprt_per_rank = length(resampled_indices) ÷ nrank
+    stock_queue = [Int[] for _ in 1:nrank]
+
+    # Assign each resampled index to its corresponding rank
+    for resampled_idx in resampled_indices
+        rank = div(resampled_idx - 1, nprt_per_rank) + 1
+        push!(stock_queue[rank], resampled_idx)
+    end
+
+    supply_vector = Float64[length(stock_queue[rank]) for rank in 1:nrank]
+    demand_vector = Float64.(fill(nprt_per_rank, nrank))
+    cost_matrix = ones(Float64, nrank, nrank)
+    for i in 1:nrank
+        cost_matrix[i, i] = 0
+    end
+
+    # Solve the optimal transport problem using the HiGHS solver
+    y = emd(supply_vector, demand_vector, cost_matrix, HiGHS.Optimizer())
```

Comment:
```suggestion
    optimizer = HiGHS.Optimizer()
    HiGHS._set_option(optimizer, "log_to_console", false)
    y = emd(supply_vector, demand_vector, cost_matrix, optimizer)
```

This will suppress the HiGHS output, though it's not idea to need to rely on what is I assume an internal interface `HiGHS._set_option` but I couldn't figure out how to get this to work previously using public API functions